service/src/main/kotlin/fi/espoo/evaka/application/ApplicationQueries.kt

```diff
@@ -566,7 +567,8 @@ fun fetchApplicationDetails(h: Handle, applicationId: UUID): ApplicationDetails?
         LEFT JOIN person g1 ON g1.id = a.guardian_id
         LEFT JOIN (
             SELECT application_id, jsonb_agg(jsonb_build_object('id', id, 'name', name, 'contentType', content_type, 'updated', updated, 'receivedAt', received_at, 'type', type)) json
-            FROM attachment GROUP BY application_id
+            FROM attachment $attachmentWhereClause
```

Comment:
This feels hacky. Is there a better way to do this?
We could filter non-citizen attachments in the CitizenController instead?

Reply:
I think this is somewhat related to the upcoming task of indicating which attachments are uploaded by who. There we must expand the current data class so that it has some indication of the uploader and then it might be reasonable to do the filtering on the code level, instead of sql. Of course that data class augmentation could be done here already but I dunno... This is fine anyway, to me at least.